aesara/graph/op.py

```diff
@@ -225,7 +225,13 @@ def make_node(self, *inputs: Variable) -> Apply:
             )
         if not all(inp.type == it for inp, it in zip(inputs, self.itypes)):
             raise TypeError(
-                f"We expected inputs of types '{str(self.itypes)}' but got types '{str([inp.type for inp in inputs])}'"
+                "Unexpected input types:\n"
+                + "\n".join(
+                    f'{i}. {">>> " if inp != out else ""}Expected {inp} - Got {out}{" <<<" if inp != out else ""}'
```

Comment:
Instead of printing all the expected/received pairs, let's print only the conflicting ones (and which number input it was, of course).

Reply:
Actually, it might help to provide context by printing the `str` representation of the node beforehand, too.

Reply:
E.g. 
```
f"Op {self} expected inputs of ..."
```

Reply:
Yeah, I like that idea

Reply:
Updated the top message with the new output format